### PR TITLE
[NEW] Enterprise Edition

### PR DESCRIPTION
**Pull request by sampaiodiego**
All hard work from @pierre-lehnen-rc <3

### CODE REVIEW

File: ee/app/livechat-enterprise/server/hooks/addDepartmentAncestors.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { callbacks } from '../../../../../app/callbacks';
+import LivechatRooms from '../../../../../app/models/server/models/LivechatRooms';
+import { LivechatDepartment } from '../../../../../app/models/server/models/LivechatDepartment';
```
Comment (renatobecker) [inline ee/app/livechat-enterprise/server/hooks/addDepartmentAncestors.js]:
```suggestion
import LivechatDepartment from '../../../../../app/models/server/models/LivechatDepartment';
```

File: ee/app/livechat-enterprise/client/route.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { AccountBox } from '../../../../app/ui-utils';
+import { livechatManagerRoutes } from '../../../../app/livechat/client/route';
+
+AccountBox.addRoute({
```
Comment (renatobecker) [inline ee/app/livechat-enterprise/client/route.js]: The Livechat Enterprise routes are being added without license verification, unlike other modules, like this:
https://github.com/RocketChat/Rocket.Chat/pull/16944/files#diff-81b0ad498000ae05840d29ce6caf4fddR6

The License verification is only inside the Templates, due to that we can face something like this:

Screen Shot 2020-03-20 at 12 17 41

I don't think it's the expected behaviour, am I right?
Why do we have different approaches among the Enterprise modules?

File: ee/i18n/en.i18n.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  "error-max-number-simultaneous-chats-reached": "The maximum number of simultaneous chats per agent has been reached.",
+  "Add_monitor": "Add monitor",
+  "Available_departments": "Available Departments",
+  "Canned Responses": "Canned Responses",
+  "Canned_Response_Removed": "Canned Response Removed",
+  "Canned_Responses_Enable": "Enable Canned Responses",
+  "Edit_Tag": "Edit Tag",
+  "Edit_Unit": "Edit Unit",
+  "Enterprise_License": "Enterprise License",
+  "Failed_to_add_monitor": "Failed to add monitor",
```
Comment (renatobecker) [inline ee/i18n/en.i18n.json]:
```suggestion
  "Failed_to_add_monitor": "The selected user is not a monitor",
```
I think we can improve it, it seems to be too technical to final users.
Maybe we can do something like we were doing on the old `Enterprise` version:
https://github.com/RocketChat/Rocket.Chat.Enterprise/blob/542717863ef8dd947f147d604d468b3befafb2ce/packages/rocketchat-enterprise-i18n/i18n/en.i18n.json#L36

Comment (sampaiodiego) [reply ee/i18n/en.i18n.json]: they're not the same.. "Failed_to_add_monitor" is being used when an unknown error happens.. so I think it's fine

File: ee/app/livechat-enterprise/client/route.js (hunk 2 above)
Comment (sampaiodiego) [reply ee/app/livechat-enterprise/client/route.js]: we can think later how the best approach for this.. we could even add the menu options but show a message about that feature being enterprise only, for example.. having decided the best approach, we can implement everywhere 😉

**Pull request merged**